pkg/cmd/addon/index.go

```diff
@@ -0,0 +1,176 @@
+package addon
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"regexp"
+
+	"github.com/apecloud/kbcli/pkg/printer"
+	"github.com/apecloud/kbcli/pkg/types"
+	"github.com/apecloud/kbcli/pkg/util"
+	"github.com/apecloud/kbcli/pkg/util/gitutil"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/klog/v2"
+)
+
+type Index struct {
+	Name string
+	URL  string
+}
+
+func init() {
+	addonDir, err := util.GetCliAddonDir()
+	if err != nil {
+		klog.V(1).ErrorS(err, "can't get the addon ")
+	}
+
+	defaultIndexDir := path.Join(addonDir, types.KubeBlocksReleaseName)
+	if _, err := os.Stat(defaultIndexDir); os.IsNotExist(err) {
+		err = gitutil.EnsureCloned(types.DefaultAddonIndexURL, defaultIndexDir)
```

Comment:
Clone github repo in `init` function is not a good choice, better to clone when run the corresponding command.